Modules/timemodule.c

```diff
@@ -2051,34 +2051,51 @@ PyInit_time(void)
 static int
 pysleep(_PyTime_t secs)
 {
-    _PyTime_t deadline, monotonic;
 #ifndef MS_WINDOWS
 #ifdef HAVE_CLOCK_NANOSLEEP
     struct timespec timeout_abs;
 #else
     struct timeval timeout;
 #endif
+    _PyTime_t deadline, monotonic;
     int err = 0;
     int ret = 0;
-#else
-    _PyTime_t millisecs;
-    unsigned long ul_millis;
-    DWORD rc;
-    HANDLE hInterruptEvent;
-#endif
 
     if (get_monotonic(&monotonic) < 0) {
         return -1;
     }
     deadline = monotonic + secs;
-#if defined(HAVE_CLOCK_NANOSLEEP) && !defined(MS_WINDOWS)
+#ifdef HAVE_CLOCK_NANOSLEEP
     if (_PyTime_AsTimespec(deadline, &timeout_abs) < 0) {
         return -1;
     }
 #endif
+#else
+    _PyTime_t millisecs;
+    DWORD ul_millis;
+    DWORD rc;
+    FILETIME FileTime;
+    LARGE_INTEGER deadline;
+    HANDLE hTimer = NULL;
+
+    GetSystemTimePreciseAsFileTime(&FileTime);
+    CopyMemory(&deadline, &FileTime, sizeof(FILETIME));
+
+    /* convert to 100 nsec unit */
+    deadline.QuadPart += (_PyTime_AsMicroseconds(secs, _PyTime_ROUND_CEILING) * 10);
+
+    hTimer = CreateWaitableTimerW(NULL, FALSE, NULL);
```

Comment:
You should close the handle somewhere, no? Maybe it can be stored in a module state: create it when the time module is created, and only close it when the time module is destroyed.

The doc says: "Use the CloseHandle function to close the handle."

Reply:
Yes, it needs to close. [Microsoft example](https://docs.microsoft.com/en-us/windows/win32/sync/using-waitable-timer-objects) was to crappy about it, i will fix it. Is that ["goto" coding-technique](https://github.com/torvalds/linux/blob/master/virt/kvm/kvm_main.c#L3150) free to use for error handling?

Reply:
> Maybe it can be stored in a module state: create it when the time module is created, and only close it when the time module is destroyed.
> 

Can you show where can i found the "constructor" and "destructor" functions of the time module? Is it ready to use in multiprocessing usage?